### PR TITLE
Added starting loss to functional train

### PR DESCRIPTION
**Pull request by jwmandeville**
Added a parameter to functional train: `starting_loss`. This is useful when tracking loss across multiple calls to the train function (eg training on multiple data splits) because it will only save the model as 'best' if the loss is lower than `starting_loss`. The behavior of `train` is unchanged if `starting_loss` isn't specified.

Change type:

- [ ] Bug fix
- [x] Feature
- [ ] Documentation

Checklist:

- [x] My code follows the style guidelines of this [project](https://docs.google.com/document/d/1jckZJe0CrWyF-IjxoO2OfBKAyKLC3Yk9Xr7UmOLBETA/edit?usp=sharing)
- [x] I have performed a self-review of my own code
- [x] I have commented my code and added docstrings to all exposed functions and classes
- [x] I have made corresponding changes to the documentation
- [x] Any relevant changes to third party licenses have been updated in the README

### CODE REVIEW

File: xt_training/utils/functional.py [hunk 1]
```diff
@@ -89,11 +91,11 @@ def train(
         for loader_name, loader in test_loaders.items():
             runner(loader, loader_name)
 
-    best_loss = 1e12
-    if val_loader:
+    best_loss = 1e12 if starting_loss is None else starting_loss
+    if val_loader and starting_loss is None:
```
Comment (timesler) [inline xt_training/utils/functional.py:95]: How about we just set the default input to `best_loss=1e12`, then we can remove line 94 and change line 95 to `if val_loader:`

**Review by timesler — CHANGES_REQUESTED** [review]

File: xt_training/utils/functional.py (hunk 1 above)
Comment (jwmandeville) [reply xt_training/utils/functional.py:95]: I've set it up this way because I don't want it to calculate the validation loss before training since I already provide the starting loss. I did this because when doing k-fold cross validation it was calculating the initial loss k times, which was slow.

I still want to provide `val_loaders` so that it can calculate the new validation loss after training though, and there wasn't any way to skip the initial validation loss while including `val_loaders`.

Maybe we should control that with a separate flag to be more clear.

Comment (timesler) [reply xt_training/utils/functional.py:95]: Ah got it. Good as it is then I reckon

**Review by timesler — APPROVED** [review]
